docs/reference/api/storage/File.md

```diff
@@ -2,12 +2,25 @@
 
 <span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_file.png)<span>File class hierarchy</span></span>
 
-[Add description here.]
+With a file system, the File class provides access to the underlying storage in a generic manner that is abstracted by the file system from any technical limitations of the underlying storage. On the storage, each file is a blob of data referenced by a file name represented as a string. By providing a file name to the `open` function in the File class, you can access the blob of data stored on disk.
```

Comment:
Query: What is this first sentence trying to say?

Reply:
I read that as:
"The File class provides access to the underlying storage of a filesystem in a generic manner. This acts as a buffer between the filesystem and any technical limitations of the physical storage medium." 

@geky feel free to correct me if I'm wrong.